src/connectors/residentadvisor.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const domain = 'https://www.residentadvisor.net';
+
+Connector.playerSelector = '.content-list';
+
+Connector.playButtonSelector = '.controls .play';
+
+// This identifies the track element depending on the parents of the play button
+function getTrackContainer() {
+	let track;
+	// Iterate through all parents until it also finds a cover art
+	$('.play.paused').parents().each((_i, parent) => {
+		if ($(parent).find('img[src^="/images/cover/"]').length > 0) {
+			track = parent;
+			return false;
+		}
+	});
+	return track;
+}
+
+function typeOfTrack(trackContainer) {
+	if ($(trackContainer).is('article')) {
+		// https://www.residentadvisor.net/tracks ("Popular tracks" at the top)
+		// https://www.residentadvisor.net/tracks/931382 ("Popular tracks" at the bottom)
+		return 'popular';
+	}
+	if ($(trackContainer).is('ul')) {
+		// https://www.residentadvisor.net/tracks ("Archived tracks" in the middle)
+		return 'archived';
+	}
+	if ($(trackContainer).is('div')) {
+		// https://www.residentadvisor.net/tracks/931382 ("Single track" in the middle)
+		return 'single';
+	}
+	if ($(trackContainer).is('li') && $(trackContainer).parent().hasClass('tracks')) {
+		// https://www.residentadvisor.net/record-label.aspx?id=15687&show=tracks (Label tracks)
+		// https://www.residentadvisor.net/dj/hotsince82/tracks (DJ tracks)
+		return 'djOrLabel';
+	}
+	if ($(trackContainer).is('li') && $(trackContainer).parent().hasClass('chart')) {
+		// https://www.residentadvisor.net/dj/secretsundaze/top10?chart=214484 (Chart tracks)
+		return 'chart';
+	}
+}
+
+Connector.getTrack = () => {
+	let track = getTrackContainer();
+	switch (typeOfTrack(track)) {
```

Comment:
Would it make more sense to call `typeOfTrack` once and bind `Connector.getTrack = () => {` for each type once?

Reply:
Are you thinking of something like https://github.com/web-scrobbler/web-scrobbler/blob/master/src/connectors/daytrotter.js where the player is initialised once when the page loads? I tried and I realised that in our case the player does not depend on the URL and we don't know which type of player will be used in advance (some pages have different types of player).

Reply:
Yes one page can have multiple different players, and one player can be on multiple pages. To avoid defining which page has which player(s), we detect the type of player depending on the context of the "pause/play" button.

1. Get the playing button
2. Get the parent player of that button
3. Detect what kind of player it is
4. Depending on the type of player, find track information